modules/lvgl/input/lvgl_encoder_input.c

```diff
@@ -64,7 +59,7 @@ static int lvgl_encoder_input_init(const struct device *dev)
 		.button_input_code = BUTTON_CODE(inst),                                            \
 	};                                                                                         \
 	static struct lvgl_common_input_data lvgl_common_input_data_##inst;                        \
-	DEVICE_DT_INST_DEFINE(inst, lvgl_encoder_input_init, NULL, &lvgl_common_input_data_##inst, \
+	DEVICE_DT_INST_DEFINE(inst, NULL, NULL, &lvgl_common_input_data_##inst,                    \
```

Comment:
how about leaving the input function there (it has the device type and may need to do something else as well in the future) and instead you only defer `lv_indev_drv_register` to `lvgl_init_input_devices`? I'm thinking that that would allow you to group all type together in an array of 'struct device' and make that code a bit less quirky.

Reply:
Unsure if I understood correctly, but e.g. `lv_indev_set_button_points(data->indev ...` depends on `data->indev` which is the result of `lv_indev_drv_register`. So I guess at least for the button this is not possible.

Reply:
Oh I see, that's unfortunate. Then I would say leave the spcific initializers in their own files, just drop the static and call them from `lvgl_init_input_devices`, basically the pattern you did here already for `lvgl_init_button_device` but for encoder and pointer as well. Just looking into leaking as few device type details out of their individual files as possible.